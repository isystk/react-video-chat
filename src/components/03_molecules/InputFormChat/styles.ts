import { css, keyframes } from '@emotion/css'

const useStyles = (theme) => css`{
  chatInput: {
    display: 'flex',
    backgroundColor: '#fff',
    overflow: 'hidden',
  },
  iconBlock: {
    width: '320px',
    height: '110px',
    backgroundColor: '#fff',
    border: '1px solid #777',
    flexWrap: 'wrap',
    position: 'absolute',
    bottom: '60px',
    left: '0',
    zIndex: '1',
    display: 'flex',
    '@media screen and (min-width:992px)': {
      left: '310px',
    },
  },
  myIcon2: {
    padding: '19px 5px',
    textAlign: 'center',
    cursor: 'pointer',
  },
  myIcon2Img: {
    width: '24px',
    height: 'auto',
    margin: '10px',
  },
  sendMsg: {
    margin: '12px !important',
  },
}`

export { useStyles }
