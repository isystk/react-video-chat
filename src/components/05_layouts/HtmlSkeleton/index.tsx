import React, { Children, FC, useState } from 'react'
import { connect } from '@/components/hoc'
import { ContainerProps, WithChildren } from 'types'
import useAppRoot from '@/stores/useAppRoot'
import Header from '@/components/04_organisms/Header'
import SideMenu from '@/components/04_organisms/SideMenu'
import Title, { TitleProps } from './Title'
import Head from 'next/head'
import { isReactElement } from '@/utils/general/object'
import { CssBaseline } from '@material-ui/core'

/** HtmlSkeleton Props */
export type HtmlSkeletonProps = WithChildren
/** Presenter Props */
export type PresenterProps = HtmlSkeletonProps & {
  title?: TitleProps['children']
  description: string
}

/** Presenter Component */
const HtmlSkeletonPresenter: FC<PresenterProps> = ({
  main,
  title,
  description,
  isMenuOpen,
  setMenuOpen,
  children,
  ...props
}) => (
  <>
    <Head>
      {/* タイトル */}
      <title>{title}</title>
      {/* favicon */}
      <link rel="icon" href="/favicon.ico" />
      {/* ブラウザテーマカラー */}
      <meta name="theme-color" content="#ffffff" />
      {/* サイト概要 */}
      <meta name="description" content={description} />
      {/* OGP 画像URL */}
      <meta property="og:image" content={'/ogp-image.png'} />
      {/* OGP タイトル */}
      <meta name="og:title" content={title} />
      {/* OGP サイト概要 */}
      <meta name="og:description" content={description} />
      {/* OGP Twitterカード */}
      <meta name="twitter:card" content="summary" />
      {/* apple ポータブル端末 アイコン */}
      <link rel="apple-touch-icon" href="/apple-touch-icon.png" />
      {/* manifest.json */}
      <link rel="manifest" href="/manifest.json" />
    </Head>
    <Context.Provider value={main}>
      <CssBaseline />
      <Header isMenuOpen={isMenuOpen} setMenuOpen={setMenuOpen} />
      <div>{children}</div>
      <SideMenu isMenuOpen={isMenuOpen} setMenuOpen={setMenuOpen} />
    </Context.Provider>
  </>
)

/** Container Component */
const HtmlSkeletonContainer: React.FC<
  ContainerProps<HtmlSkeletonProps, PresenterProps>
> = ({ presenter, children, ...props }) => {
  let title: TitleProps['children'] | undefined = undefined
  const main = useAppRoot()
  const [isMenuOpen, setMenuOpen] = useState(false)

  if (!main) return <></>

  const APP_NAME = process.env.APP_NAME

  children = Children.map(children, (child) =>
    isReactElement(child) && child.type === Title
      ? (title = `${child.props.children} | ${APP_NAME}`) && undefined
      : React.cloneElement(child)
  )
  if (!title) title = process.env.APP_NAME

  const description = `${APP_NAME} is an application that allows you to casually chat with your friends!`
  return presenter({
    main,
    title,
    description,
    isMenuOpen,
    setMenuOpen,
    children,
    ...props,
  })
}

export const Context = React.createContext(null)

export default connect<HtmlSkeletonProps, PresenterProps>(
  'HtmlSkeleton',
  HtmlSkeletonPresenter,
  HtmlSkeletonContainer
)

// Sub Component
export type { TitleProps }
export { Title }
