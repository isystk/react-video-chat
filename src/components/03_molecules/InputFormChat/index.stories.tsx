import { Meta, Story } from '@storybook/react'
import React from 'react'
import InputFormChat from './index'
import MainService from '@/services/main'
import { Context } from '@/components/05_layouts/HtmlSkeleton'

export default {
  title: '03_molecules/InputFormChat',
  component: InputFormChat,
} as Meta

const Template: Story = (props) => {
  const main = new MainService(() => ({}))

  return (
    <Context.Provider value={main}>
      <InputFormChat />
    </Context.Provider>
  )

}

export const Primary = Template.bind({})
Primary.storyName = 'プライマリ'
Primary.args = {}
