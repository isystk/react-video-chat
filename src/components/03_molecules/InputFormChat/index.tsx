import TextField from '@material-ui/core/TextField'
import React, { FC, useCallback, useContext, useEffect, useState } from 'react'
import Button from '@material-ui/core/Button'
import { ContainerProps, WithChildren } from 'types'
import { useStyles } from './styles'
import { connect } from '@/components/hoc'
import TagFacesIcon from '@material-ui/icons/TagFaces'
import AttachFileIcon from '@material-ui/icons/AttachFile'
import SendIcon from '@material-ui/icons/Send'
import { Stamps } from '@/services/Chat'
import { Context } from '@/components/05_layouts/HtmlSkeleton'
import MainService from '@/services/main'

/** InputFormChatProps Props */
export type InputFormChatProps = WithChildren
/** Presenter Props */
export type PresenterProps = InputFormChatProps & {
  classes
  main
  Stamps
  label
  setMessage
  message
  setIsComposed
  isComposed
  initializeLocalPeer
  isHover
  handleMouseEnter
  handleMouseLeave
}

/** Presenter Component */
const InputFormChatPresenter: FC<PresenterProps> = ({
  classes,
  main,
  Stamps,
  label,
  setMessage,
  message,
  setIsComposed,
  isComposed,
  initializeLocalPeer,
  isHover,
  handleMouseEnter,
  handleMouseLeave,
  ...props
}) => (
  <>
    <div className={classes.chatInput}>
      <div
        className={classes.myIcon2}
        onMouseEnter={handleMouseEnter}
        onMouseLeave={handleMouseLeave}
      >
        <TagFacesIcon color="action" />
        {isHover && (
          <div className={classes.iconBlock}>
            {_.map(Stamps, (stamp, key) => (
              <div
                key={key}
                onClick={(e) =>
                  main.chanels[main.selectChanelId].chat.sendStamp(key)
                }
              >
                <img src={stamp} alt="" className={classes.myIcon2Img} />
              </div>
            ))}
          </div>
        )}
      </div>
      <div className={classes.myIcon2}>
        <a href="#" onClick={() => alert('作成中')}>
          <AttachFileIcon color="action" />
        </a>
      </div>
      <TextField
        className={classes.sendMsg}
        autoFocus
        fullWidth
        label={label}
        name="message"
        size="small"
        onChange={(e) => setMessage(e.target.value)}
        onCompositionEnd={() => setIsComposed(false)}
        onCompositionStart={() => setIsComposed(true)}
        onKeyDown={async (e) => {
          if (isComposed) return
          if (e.target.value === '') return
          if (e.key === 'Enter') await initializeLocalPeer(e)
        }}
        required
        value={message}
        variant="outlined"
      />
      <Button onClick={async (e) => await initializeLocalPeer(e)} type="submit">
        <SendIcon color="primary" />
      </Button>
    </div>
  </>
)

/** Container Component */
const InputFormChatContainer: React.FC<
  ContainerProps<InputFormChatProps, PresenterProps>
> = ({ presenter, children, ...props }) => {
  const main = useContext<MainService | null>(Context)
  if (!main) return <></>
  const classes = useStyles()
  const label = 'メッセージを入力してください'
  const [disabled, setDisabled] = useState(true)
  const [message, setMessage] = useState('')
  const [isComposed, setIsComposed] = useState(false)
  const [isHover, setIsHover] = useState(false)
  const handleMouseEnter = () => {
    setIsHover(true)
  }
  const handleMouseLeave = () => {
    setIsHover(false)
  }

  useEffect(() => {
    const disabled = message === ''
    setDisabled(disabled)
  }, [message])

  const initializeLocalPeer = useCallback(
    async (e) => {
      e.persist()
      await main.chanels[main.selectChanelId].chat.sendChat(message)
      setMessage('')
      e.preventDefault()
    },
    [message, main]
  )

  return presenter({
    children,
    main,
    classes,
    Stamps,
    label,
    setMessage,
    message,
    setIsComposed,
    isComposed,
    initializeLocalPeer,
    isHover,
    handleMouseEnter,
    handleMouseLeave,
    ...props,
  })
}

export default connect<InputFormChatProps, PresenterProps>(
  'InputFormChat',
  InputFormChatPresenter,
  InputFormChatContainer
)
