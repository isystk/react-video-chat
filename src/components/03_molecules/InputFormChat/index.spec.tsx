import React from 'react'
import renderer from 'react-test-renderer'
import InputFormChat from './index'
import '@testing-library/jest-dom/extend-expect'
import { Context } from '@/components/05_layouts/HtmlSkeleton'
import MainService from '@/services/main'

describe('InputFormChat', () => {
  it('Match Snapshot', () => {
    const main = new MainService(() => ({}))

    const component = renderer.create(
      <Context.Provider value={main}>
        <InputFormChat />
      </Context.Provider>
    )
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
