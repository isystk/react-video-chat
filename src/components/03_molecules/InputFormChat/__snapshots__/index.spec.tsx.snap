// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputFormChat Match Snapshot 1`] = `
<div
  className="css-1ehe1el"
>
  <div
    className="css-p9zv6y"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-colorAction MuiSvgIcon-fontSizeMedium css-11pbyhm-MuiSvgIcon-root"
      data-testid="TagFacesIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm3.5-9c.83 0 1.5-.67 1.5-1.5S16.33 8 15.5 8 14 8.67 14 9.5s.67 1.5 1.5 1.5zm-7 0c.83 0 1.5-.67 1.5-1.5S9.33 8 8.5 8 7 8.67 7 9.5 7.67 11 8.5 11zm3.5 6.5c2.33 0 4.31-1.46 5.11-3.5H6.89c.8 2.04 2.78 3.5 5.11 3.5z"
      />
    </svg>
  </div>
  <div
    className="css-p9zv6y"
  >
    <a
      href="#"
      onClick={[Function]}
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-colorAction MuiSvgIcon-fontSizeMedium css-11pbyhm-MuiSvgIcon-root"
        data-testid="AttachFileIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M16.5 6v11.5c0 2.21-1.79 4-4 4s-4-1.79-4-4V5c0-1.38 1.12-2.5 2.5-2.5s2.5 1.12 2.5 2.5v10.5c0 .55-.45 1-1 1s-1-.45-1-1V6H10v9.5c0 1.38 1.12 2.5 2.5 2.5s2.5-1.12 2.5-2.5V5c0-2.21-1.79-4-4-4S7 2.79 7 5v12.5c0 3.04 2.46 5.5 5.5 5.5s5.5-2.46 5.5-5.5V6h-1.5z"
        />
      </svg>
    </a>
  </div>
  <div
    className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-1t3kwb2 css-wb57ya-MuiFormControl-root-MuiTextField-root"
    onCompositionEnd={[Function]}
    onCompositionStart={[Function]}
    onKeyDown={[Function]}
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined css-1pysi21-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink={false}
      htmlFor=":r0:"
      id=":r0:-label"
    >
      メッセージを入力してください
      <span
        aria-hidden={true}
        className="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
      >
         
        *
      </span>
    </label>
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={true}
        className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
        disabled={false}
        id=":r0:"
        name="message"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={true}
        type="text"
        value=""
      />
      <fieldset
        aria-hidden={true}
        className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="css-yjsfm1"
        >
          <span>
            メッセージを入力してください
             
            *
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="submit"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-ci5apu-MuiSvgIcon-root"
      data-testid="SendIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M2.01 21 23 12 2.01 3 2 10l15 2-15 2z"
      />
    </svg>
  </button>
</div>
`;
